node/src/checkpointers.test.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2021 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as checkpointers from './checkpointers';
+import path from 'path';
+import fs from 'fs';
+import { CheckpointerState, FileCheckPointer } from './filecheckpointer';
+import { InmemoryCheckPointer } from './inmemorycheckpointer';
+
+async function writeToFile(
+    checkpointerstate: CheckpointerState,
+    path: string
+): Promise<void> {
+    const data = Buffer.from(JSON.stringify(checkpointerstate));
+    await fs.promises.writeFile(path, data);
+}
+
+const dir = './testUtils';
+const fileName = 'checkpoint.json';
+
+const checkPointerFilePath = path.join(dir, fileName);
+
+function getInmemoryInstance():Promise<InmemoryCheckPointer>{
+    return Promise.resolve(checkpointers.inMemory());
+}
+
+async function getFileCheckpointerInstance():Promise<FileCheckPointer>{
+    const x = await checkpointers.file(checkPointerFilePath);
+    return x
+}
+
+const checkpointerTypes = [
+    {   getInstance: getInmemoryInstance,
+        description: 'Inmemory checkpointer',
+    },
+    {
+        getInstance: getFileCheckpointerInstance,
+        description: 'File checkpointer',
+    },
+];
+
+checkpointerTypes.forEach(checkpointer=>{
+    describe(`${checkpointer.description}`, () => {
+        beforeEach(async() => {
+            if (!fs.existsSync(dir)){
+                fs.mkdirSync(dir);
+            }
+            await fs.promises.writeFile(checkPointerFilePath, '');
+        })
+
+        afterAll( ()=>{
+            if (fs.existsSync(dir)){
+                fs.rmSync(dir, { recursive: true, force: true });
+            }
+        })
```

Comment:
It would be good not to mess with the filesystem unless it's the file checkpointer being tested